app/templates/ui/app/detail/detail.controller.js

```diff
@@ -40,7 +42,41 @@
 
     angular.extend(ctrl, {
       doc : doc.data,
-      uri : uri
+      uri : uri,
+      delete: deleteFunc
     });
+
+    function deleteFunc() {
+
+      $uibModal.open({
+        controller: ['$uibModalInstance', function($uibModalInstance) {
+          var ctrl = this;
+
+          ctrl.confirm = function(answer) {
+            if (answer === true) {
+              MLRest.deleteDocument (uri).then(function(response) {
+                // create a toast with settings:
+                toast.create({
+                  className: 'warning',
+                  content: 'Deleted ' + uri,
+                  dismissOnTimeout: true,
+                  timeout: 2000,
+                  onDismiss: function () {
+                    //redirect to search page
+                    $state.go('root.search');
+                  }
+                });
+              });
+            }
+            return $uibModalInstance.close();
+          };
+        }],
+        controllerAs: 'ctrl',
+        templateUrl: '/app/detail/confirm-modal.html',
+        backdrop: 'static',
+        keyboard: false
+      });
+    }
+
```

Comment:
Ah right, I had something simpler in mind when I mentioned popover: https://angular-ui.github.io/bootstrap/#/popover. Showing a modal is perhaps a bit overkill. I used that myself as well before, but liked the popover better when I saw it being used instead recently on some ordinary webpage.